### PR TITLE
platform: Add nuclei ux600 platform

### PR DESCRIPTION
**Pull request by fanghuaqi**
This PR is used to adding support for Nuclei UX600 RISC-V 64bit Core.

> Nuclei UX600 is a riscv 64bit core developed by Nuclei System Technology, see https://nucleisys.com/product.php

### CODE REVIEW

File: lib/utils/sys/nuclei_timer.c [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+int nuclei_timer_cold_timer_init(unsigned long base, u32 hart_count,
+			  bool has_64bit_mmio)
+{
+	/* Figure-out RV_TIMER Time register address */
+	nuclei_timer_time_hart_count		= hart_count;
+	nuclei_timer_time_base			= (void *)base;
+	nuclei_timer_time_val			= (u64 *)(nuclei_timer_time_base + 0x0);
+	nuclei_timer_time_cmp			= (u64 *)(nuclei_timer_time_base + 0x8);
+
```
Comment (atishp04) [inline lib/utils/sys/nuclei_timer.c:149]: It looks like this timer block is identical to a clint block except the offset for timeval and timecmp.
If that's the case, generic clint can be improved in such a way that, ux600 can use clint block as it is.

File: lib/utils/sys/objects.mk [hunk 2]
```diff
@@ -7,5 +7,6 @@
 #   Anup Patel <anup.patel@wdc.com>
 #
 
+libsbiutils-objs-y += sys/nuclei_timer.o
```
Comment (atishp04) [inline lib/utils/sys/objects.mk:10]: As nuclei_timer is very similar to clint, we should try to reuse the code.

File: platform/nuclei/ux600/platform.c [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	return plic_warm_irqchip_init(hartid, (hartid) ? (2 * hartid - 1) : 0,
+				      (hartid) ? (2 * hartid) : -1);
+}
+
+static int ux600_ipi_init(bool cold_boot)
+{
+	int rc;
+
+	if (cold_boot) {
+		rc = nuclei_timer_cold_ipi_init(UX600_RV_TIMER_ADDR + 0xffc, UX600_HART_COUNT);
```
Comment (atishp04) [inline platform/nuclei/ux600/platform.c:115]: please define a macro for 0xffc as well.

File: lib/utils/sys/nuclei_timer.c (hunk 1 above)
Comment (fanghuaqi) [reply lib/utils/sys/nuclei_timer.c:149]: Thanks for your review, I will check and see how to easily reuse with clint code.

**Comment by fanghuaqi** [comment]
Hi @atishp04 , I talked with our hw engineer, our nuclei timer will also be compabable with clint  timer, I also update our opensbi portable code, I will close this PR, and use new PR #162 to  track it.

Thanks
Huaqi

**Pull request closed without merging**